src/main/java/homework_3/ImmutableWorker.java

```diff
@@ -0,0 +1,114 @@
+package homework_3;
+
+import jdk.nashorn.internal.ir.annotations.Immutable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * In general immutable class means that once an object is created, we cannot change its content, so:
+ * Immutable class should be final.
+ * Its fields should be private final.
+ * There should be getters but no setters.
+ * For link objects there should be a deep copy in the constructor.
```

Comment:
Constructor always creates new, isn't it?

Reply:
yes, it's true